flake.nix

```diff
@@ -35,9 +34,8 @@
         checks.prettier =
           let
             checkFormatCommand = packageJson.scripts.check-format;
-            buildInputs = self.packages.${system}.default.buildInputs;
           in
-          pkgs.runCommand "check-ts-format" { inherit buildInputs; } ''
+          pkgs.runCommand "check-ts-format" { buildInputs = with pkgs.nodePackages; [ prettier ]; } ''
```

Comment:
I think this results in different versions of `prettier` for the node environment and this check.

Reply:
This is a bit problematic 🥲
Contrary to many other language specific package managers, `npm` treats development tools as dependencies and pins them with a version in `package-lock.json`.
AFAIK there is no way to get the dependencies from `buildNpmPackage` as derivation.
Other solutions like pinning versions of tools like `prettier` so that the versions from `nixpkgs` in `flake.lock` and `package-lock.json` are the same, is not really practicable.

But then again, I'm not sure how much it matters in practise: In other languages such as Rust, there is always a potential version discrepancy in development tools between nix builds and non-nix builds, as these tools (such as `clippy`) are not pinned. But if that is a hard requirement, then staying with `node2nix` is most probably the better choice.

Reply:
IMHO, having inferior situations elsewhere, is a poor argument.

At least within `buildNpmPackage`, dependencies must be available:

```nix
{
  checks.prettier = self.packages.${system}.default.overrideAttrs(oldAttrs: {
    name = "${oldAttrs.name}-prettier";
    buildPhase = "npm run check-format";
  });
}
```

Reply:
Alternatively, you could just add another `checkFormat` phase to the `default` package derivation 🤷🏻‍♂️

Reply:
Uh, wasn't about inferior situations, just was way to focused on on this "getting dependency out of `buildNpmPackage`" 🤦 
But of course rewriting the check is the solution here 😅 
Will fix.